include/grpc++/impl/codegen/thrift_serializer.h

```diff
@@ -200,11 +200,9 @@ class ThriftSerializer {
   bool serialize_version_;
 
   void prepare() {
-
-    buffer_.reset(new TMemoryBuffer());
-
+    buffer_ = boost::make_shared<TMemoryBuffer>(*(new TMemoryBuffer()));
```

Comment:
This defeats the purpose of `make_shared`
It should just be
`buffer_ = boost::make_shared<TMemoryBuffer>()`

`make_shared` does the new for you.